File: leak/poller.py
#!/usr/bin/python3

import sys, signal
#from PySide import QtCore, QtNetwork
from PyQt4 import QtCore, QtGui, QtNetwork
QtCore.Signal = QtCore.pyqtSignal
QtCore.Slot = QtCore.pyqtSlot

import rpc

class Poller(QtCore.QObject):

    url = QtCore.QUrl("http://localhost:5000/")

    def __init__(self, parent=None):
        super(Poller,self).__init__(parent)
        self.manager = rpc.RPCManager()

    def start(self):
        self.reply = self.manager.call('getinfo')
        self.reply.finished.connect(self.readReply)
        self.reply.error.connect(self.error)

    def readReply(self):
        text = self.reply.readAll()
        self.reply.deleteLater()
        #self.reply = None # culprit??
        QtCore.QTimer.singleShot(10, self.start)

    @QtCore.Slot(QtNetwork.QNetworkReply.NetworkError)
    def error(self, err):
        sys.stderr.write('Error: {}\n'.format(self.sender().errorString()))
        self.reply.finished.disconnect()
        self.reply.deleteLater()
        QtCore.QTimer.singleShot(10, self.start)

signal.signal(signal.SIGINT, signal.SIG_DFL)
app = QtGui.QApplication(sys.argv)
p = Poller()
p.start()
app.exec_()

File: tools/leak/poller.cpp
#include <QtCore>
#include <QtNetwork>

class Poller : public QObject
{ Q_OBJECT

public slots:
	void start() {
		QNetworkRequest request(m_url);
		m_reply = m_manager.get(QNetworkRequest(QUrl("http://localhost:5000/")));
		connect(m_reply, SIGNAL(finished()), this, SLOT(readReply()));
		connect(m_reply, SIGNAL(error(QNetworkReply::NetworkError)),
				this, SLOT(error(QNetworkReply::NetworkError)));
	}

	void readReply() {
		m_reply->readAll();
		//m_reply->deleteLater();
		QTimer::singleShot(10, this, SLOT(start()));
	}

	void error(QNetworkReply::NetworkError) {
		m_reply->disconnect();
		qDebug() << "Error:" << m_reply->errorString();
		m_reply->deleteLater();
		qApp->quit();
	}

private:
	QNetworkAccessManager m_manager;
	QUrl m_url;
	QNetworkReply *m_reply;
};

int main (int argc, char *argv[]) {
	QCoreApplication app(argc, argv);
	Poller p;
	p.start();
	return app.exec();
}

#include "poller.moc"
